.storybook/custom-decorators/toast.decorator.js

```diff
@@ -1,18 +1,25 @@
 import { useEffect } from "@storybook/client-api";
 import { showToast } from "../../src/toasts/toasts.js";
 
-function getButton() {
-  return document.querySelectorAll(".toast-clicked");
-}
+let count = 0;
 
 export const withToast = storyFn => {
   useEffect(() => {
-    getButton().forEach(tab =>
-      tab.addEventListener(
-        "click",
-        showToast({ description: "This is a toast" })
-      )
-    );
-  }, []);
+    const toastEls = document.querySelectorAll(".toast-clicked");
+    toastEls.forEach(toastEl => {
+      toastEl.addEventListener("click", toastClickHandler);
+    });
+
+    return () => {
+      toastEls.forEach(toastEl => {
+        toastEl.removeEventListener("click", toastClickHandler);
+      });
+    };
+  });
+
   return storyFn();
 };
+
+function toastClickHandler() {
+  showToast({ description: `This is a toast ${count++}` });
```

Comment:
The count makes it easier to understand how multiple toasts interact with each other.